e2e/e2e_test.go

```diff
@@ -113,22 +178,138 @@ func Test_E2E_Token_Transfer(t *testing.T) {
 	waitForTransaction(setupEnv, txHash, t)
 
 	// Step 6 - Validate Token balances
-	validateWrappedAssetBalance(setupEnv, setupEnv.TokenID.String(), big.NewInt(mintAmount), wrappedBalanceBefore, setupEnv.Receiver, t)
+	verifyWrappedAssetBalance(setupEnv, setupEnv.TokenID.String(), big.NewInt(mintAmount), wrappedBalanceBefore, setupEnv.EthReceiver, t)
 
 	// Step 7 - Verify Database records
-	expectedTxRecord := prepareExpectedTransfer(
+	expectedTxRecord := util.PrepareExpectedTransfer(
 		setupEnv.Clients.RouterContract,
 		transactionResponse.TransactionID,
 		setupEnv.TokenID.String(),
-		strconv.FormatInt(amount, 10),
-		setupEnv.Receiver.String(),
+		strconv.FormatInt(tinyBarAmount, 10),
+		setupEnv.EthReceiver.String(),
 		database.ExpectedStatuses{
 			Status:          entity_transfer.StatusCompleted,
 			StatusSignature: entity_transfer.StatusSignatureMined,
 		}, t)
 
 	// Step 8 - Verify Database Records
-	verifyDatabaseRecords(setupEnv.DbValidation, expectedTxRecord, strconv.FormatInt(mintAmount, 10), receivedSignatures, t)
+	verifyTransferRecordAndSignatures(setupEnv.DbValidator, expectedTxRecord, strconv.FormatInt(mintAmount, 10), receivedSignatures, t)
+}
+
+func validateReceiverAccountBalance(setup *setup.Setup, beforeHbarBalance hedera.AccountBalance, asset string, t *testing.T) {
+	afterHbarBalance := util.GetHederaAccountBalance(setup.Clients.Hedera, setup.Clients.Hedera.GetOperatorAccountID(), t)
+
+	var beforeTransfer uint64
+	var afterTransfer uint64
+
+	switch asset {
+	case constants.Hbar:
+		beforeTransfer = uint64(beforeHbarBalance.Hbars.AsTinybar())
+		afterTransfer = uint64(afterHbarBalance.Hbars.AsTinybar())
+	case constants.Token:
+		beforeTransfer = beforeHbarBalance.Token[setup.TokenID]
+		afterTransfer = afterHbarBalance.Token[setup.TokenID]
+	default:
+		t.Fatalf("Asset [%s] is not supported", asset)
+	}
+
+	if afterTransfer-beforeTransfer != receiveAmount {
+		t.Fatalf("[%s] Expected %s balance after - [%d], but was [%d]", setup.Clients.Hedera.GetOperatorAccountID(), asset, afterTransfer, beforeTransfer)
+	}
+}
+
+func validateBurnEvent(setupEnv *setup.Setup, txHash common.Hash, expectedRouterBurn *routerContract.RouterBurn, t *testing.T) string {
+	fmt.Println(fmt.Sprintf("[%s] Waiting for Burn Transaction Receipt.", txHash))
+	txReceipt, err := setupEnv.Clients.EthClient.WaitForTransactionReceipt(txHash)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Println(fmt.Sprintf("[%s] Burn Transaction mined and retrieved receipt.", txHash))
```

Comment:
it would be better to wait for the Burn  Transaction in the function where you send it and not here. The logic there is responsible for sending the burn TX and should be responsible for waiting for its success. When you get to the point of validating the event, you should know that the TX was mined